### ISSUE TITLE
Windows 10 & 11 not booting in quickemu

### ISSUE THREAD

**Issue by PPKNexus** [quickemu-project/quickemu#609]
This the same problem dealt with in previous posts. However, the solution presented FS0:\EFI\BOOT\BOOTX64.EFI did not work.

![WindowsQuickemu2](https://user-images.githubusercontent.com/121834217/210284939-560adda4-7687-40e0-8bd7-a9d9cda73c36.png)

After following the instructions FS0:\EFI\BOOT\BOOTX64.EFI

![WindowsQuickemu1](https://user-images.githubusercontent.com/121834217/210284970-96e31e4c-b314-4dcc-9139-a3e63510aa02.png)

I press any key, even spam it, and this the result.

Host:
ROG Strix AMD Ryzen 1700
Xubuntu 20.04
5.15.0-56-lowlatency

**Comment by playingwithpy** [comment]
Hello all. I would like to report that I am also having this issue with Windows 10 & 11 not booting. Two VMs (10 and 11) that were previously working with no issue all of a sudden just won't boot. But my boot screen has a different message than @PPKNexus. 

![Screenshot from 2023-01-27 17-44-12](https://user-images.githubusercontent.com/54438192/215221827-9dd192d1-fa59-4b86-98de-786fcd4248d7.png)

Could the root cause be the same?

I have tried:
1. Restoring from a snapshot of an earlier known working state of the VM.
2. Running the config from command line as sudo and as normal user.
3. running ```echo 1 | sudo tee /sys/module/kvm/parameters/ignore_msrs``` because it said to do so in the terminal output in a warning about unhandled model specific registers.

There was also the message: ```swtpm: Could not open UnixIO socket: Permission denied``` in the output.

This is mentioned in [issue 487](https://github.com/quickemu-project/quickemu/issues/487) but I don't know if its related to this issue. 

```socat``` is installed on my system. 

Any guidance is much appreciated. Thank you for your work on this excellent tool.

**Comment by asininemonkey** [comment]
Having a similar issue. Removing `secureboot="on"` from my `.conf` files allows the VMs to start. Same behaviour when installing fresh VMs.

Not quite sure why SeaBIOS is being problematic.

**Comment by suuuehgi** [comment]
The screenshot clearly says that no bootable device was found.

You get the same message slightly clearer utilizing `boot="legacy"`

```log
No bootable device.
```

The README states:

>quickget can not download Windows 10 and Windows 11 automatically, but does automatically create an optimised virtual machine configuration that you can just add an Windows .iso image to. 

And indeed `quickget windows 10` downloads just the `virtio-win.iso` image, not a `windows-10.iso`.

So, you need to [download](https://www.microsoft.com/en-us/software-download/windows10ISO) the image separately and adapt your `iso` parameter.

E.g.

```
iso="/path/to/Win10_22H2_EnglishInternational_x64.iso"
```

**Comment by playingwithpy** [comment]
Thanks for the replies. Yes I understand this about the ```.iso``` but these were not new machines I was trying to create.

Rather, they were machines that were already installed and working for months without issue until all of a sudden I started getting this no bootable media error, having made no changes to the ```.conf``` and having the correct ```.iso``` images with the matching name in the in the ```.conf```.

Sometimes I would get the windows troubleshooting screen that comes up when you have an unbootable installation but usually I would just get the "no bootable media" screen and then the window just closed after about 20 seconds.
Both VMs had been setup using quickget several months back without issue. 

UPDATE: In my case the issue resolved itself after the most recent update was pushed out. I simply deleted both VMs and started over using quickget as suggested to get the ```virtio-win.iso```  and then using the same .```iso``` from the old VMs, taking care to make sure ```.conf``` was pointing to the right file name.

**Comment by LightningRhino** [comment]
Can't install windows 11 on my side as well.
NixOS 22.11, quickemu 4.6. socat is installed.
I have manually put the windows-11.iso in the appropriate directory as advised.
But when I start with "quickemu --vm windows-11.conf" I can see the Tiano Core screen quickly switching to the SeaBIOS screen stating "no bootable device".

Output of the starting command:

> Quickemu 4.6 using /nix/store/2zy6340qc3ms8bnvw6v7gqb8hj6cw73b-qemu-7.2.0/bin/qemu-system-x86_64 v7.2.0
>  - Host:     "NixOS 22.11 (Raccoon)" running Linux 6.1 (nixos)
>  - CPU:      AMD Ryzen 7 5700X 8-Core Processor
>  - CPU VM:   1 Socket(s), 4 Core(s), 2 Thread(s), 8G RAM
>  - MSR:      WARNING! Ignoring unhandled Model-Specific Registers is disabled.
> 
>              echo 1 | sudo tee /sys/module/kvm/parameters/ignore_msrs
> 
>              If you are unable to run macOS or Windows VMs then run the above 👆
>              This will enable ignoring of unhandled MSRs until you reboot the host.
>              You can make this change permenant by running: 'quickemu --ignore-msrs-always'
>  - BOOT:     EFI (Windows), OVMF (/nix/store/jb2kmcf1j2198pldwcpys82f7zsnkn5q-OVMF-202211-fd/FV/OVMF_CODE.fd), SecureBoot (on).
>  - Disk:     windows-11/disk.qcow2 (64G)
>              Looks unused, booting from windows-11/windows-11.iso
>  - Boot ISO: windows-11/windows-11.iso
>  - CD-ROM:   windows-11/virtio-win.iso
>  - Display:  SDL, virtio-vga-gl, GL (on), VirGL (on)
>  - ssh:      On host:  ssh user@localhost -p 22220
>  - WebDAV:   On guest: dav://localhost:9843/
>  - TPM:      windows-11/windows-11.swtpm-sock (12354)
>  - Network:  User (virtio-net)
>  - Monitor:  On host:  nc -U "windows-11/windows-11-monitor.socket"
>              or     :  socat -,echo=0,icanon=0 unix-connect:windows-11/windows-11-monitor.socket
>  - Serial:   On host:  nc -U "windows-11/windows-11-serial.socket"
>              or     :  socat -,echo=0,icanon=0 unix-connect:windows-11/windows-11-serial.socket
>  - Process:  Starting windows-11.conf as windows-11 (12356)
>  - Sending:  sendkey ret


A MacOS vm is working flawlessy by the way.
Thanks for the support.

**Comment by foolnotion** [comment]
> Can't install windows 11 on my side as well. NixOS 22.11, quickemu 4.6. socat is installed. I have manually put the windows-11.iso in the appropriate directory as advised. But when I start with "quickemu --vm windows-11.conf" I can see the Tiano Core screen quickly switching to the SeaBIOS screen stating "no bootable device".
> 
> Output of the starting command:
> 
> > Quickemu 4.6 using /nix/store/2zy6340qc3ms8bnvw6v7gqb8hj6cw73b-qemu-7.2.0/bin/qemu-system-x86_64 v7.2.0
> > 
> > * Host:     "NixOS 22.11 (Raccoon)" running Linux 6.1 (nixos)
> > * CPU:      AMD Ryzen 7 5700X 8-Core Processor
> > * CPU VM:   1 Socket(s), 4 Core(s), 2 Thread(s), 8G RAM
> > * MSR:      WARNING! Ignoring unhandled Model-Specific Registers is disabled.
> >   ```
> >         echo 1 | sudo tee /sys/module/kvm/parameters/ignore_msrs
> >   
> >         If you are unable to run macOS or Windows VMs then run the above 👆
> >         This will enable ignoring of unhandled MSRs until you reboot the host.
> >         You can make this change permenant by running: 'quickemu --ignore-msrs-always'
> >   ```
> > * BOOT:     EFI (Windows), OVMF (/nix/store/jb2kmcf1j2198pldwcpys82f7zsnkn5q-OVMF-202211-fd/FV/OVMF_CODE.fd), SecureBoot (on).
> > * Disk:     windows-11/disk.qcow2 (64G)
> >   Looks unused, booting from windows-11/windows-11.iso
> > * Boot ISO: windows-11/windows-11.iso
> > * CD-ROM:   windows-11/virtio-win.iso
> > * Display:  SDL, virtio-vga-gl, GL (on), VirGL (on)
> > * ssh:      On host:  ssh user@localhost -p 22220
> > * WebDAV:   On guest: dav://localhost:9843/
> > * TPM:      windows-11/windows-11.swtpm-sock (12354)
> > * Network:  User (virtio-net)
> > * Monitor:  On host:  nc -U "windows-11/windows-11-monitor.socket"
> >   or     :  socat -,echo=0,icanon=0 unix-connect:windows-11/windows-11-monitor.socket
> > * Serial:   On host:  nc -U "windows-11/windows-11-serial.socket"
> >   or     :  socat -,echo=0,icanon=0 unix-connect:windows-11/windows-11-serial.socket
> > * Process:  Starting windows-11.conf as windows-11 (12356)
> > * Sending:  sendkey ret
> 
> A MacOS vm is working flawlessy by the way. Thanks for the support.

Same issue here:
```
Quickemu 4.6 using /nix/store/jghgb9z5gkmfywxzjz5n1zplwymapp2i-qemu-7.2.0/bin/qemu-system-x86_64 v7.2.0
 - Host:     "NixOS 23.05 (Stoat)" running Linux 6.2 (chemalle)
 - CPU:      12th Gen Intel(R) Core(TM) i7-1265U
 - CPU VM:   1 Socket(s), 2 Core(s), 2 Thread(s), 8G RAM
 - BOOT:     EFI (Windows), OVMF (/nix/store/cs2bj0q8k977v2wcyj9gz701c4nzsl0w-OVMF-202211-fd/FV/OVMF_CODE.fd), SecureBoot (on).
 - Disk:     windows-11/disk.qcow2 (64G)
             Looks unused, booting from windows-11/windows-11.iso
 - Boot ISO: windows-11/windows-11.iso
 - CD-ROM:   windows-11/virtio-win.iso
 - Display:  SDL, virtio-vga-gl, GL (on), VirGL (on)
 - ssh:      On host:  ssh user@localhost -p 22220
 - WebDAV:   On guest: dav://localhost:9843/
 - TPM:      windows-11/windows-11.swtpm-sock (8704)
 - Network:  User (virtio-net)
 - Monitor:  On host:  nc -U "windows-11/windows-11-monitor.socket"
             or     :  socat -,echo=0,icanon=0 unix-connect:windows-11/windows-11-monitor.socket
 - Serial:   On host:  nc -U "windows-11/windows-11-serial.socket"
             or     :  socat -,echo=0,icanon=0 unix-connect:windows-11/windows-11-serial.socket
 - Process:  Starting windows-11.conf as windows-11 (8706)
 - Sending:  sendkey ret
 ```

"No bootable device"

**Comment by marius851000** [comment]
Also using NixOS. I got the ISO to load by disabling secureboot, but then, it get stuck on a blue screen, and the viewer close after bout 30 second.

**Comment by bbigras** [comment]
I have a win 11 using quickemu on nixos. I think I ran `echo 1 | sudo tee /sys/module/kvm/parameters/ignore_msrs` and I'm on nixos-unstable.

**Comment by nanonerd-io** [comment]
Just tried doing a fresh install of Windows 11, and I got the exact same error message as the OP. I upgraded to Quickemu 4.7 but to no avail. Here's what worked for me:

Launch the VM using SPICE display with the following command
`quickemu --vm windows-11.conf --display spice`

You will get the aforementioned error. You need to pass through your physical keyboard via SPICE. I'm using the Spicy client on Arch Linux, so select "Input >> Select USB devices for Redirection". Then select your physical keyboard to pass through

At the SHELL prompt, type the following then press enter
FS0:\EFI\BOOT\BOOTX64.EFI

Hit any key to boot from the virtual CD/DVD (the Windows ISO)

Windows 11 literally just finished installing no problems. Hope this helps others as I was stuck on this for a couple of hours

**Comment by LightningRhino** [comment]
Even with quickemu 4.8 same error trying to install windows 11 with 'quickemu --vm windows-11.conf':
SeaBIOS screen stating 'no bootable device'.

**Comment by me00001** [comment]
> EDIT: I also had to disable `secureboot="on"`

Same here, NixOS unstable 
CC Maintainer @FedX-sudo

**Comment by FedX-sudo** [comment]
If I had a guess, it's related to this bug: https://github.com/NixOS/nixpkgs/issues/250849. 

TL;DR:  @omernaveedxyz determined that there appears to be a regression in the Linux kernel that breaks MacOS virtualization on AMD. Given almost everyone reporting this issue (except @foolnotion) are using AMD, I suspect whatever caused the linked bug, is the root cause here.

**Comment by me00001** [comment]
> If I had a guess, it's related to this bug: [NixOS/nixpkgs#250849](https://github.com/NixOS/nixpkgs/issues/250849).

thanks for reply. it could be related. but it's not the same issue because it's has been fixed with kernel (6.1.49), mine kernel is (6.1.51) .

**Comment by flexiondotorg** [comment]
Fixed via #836

**Comment by galemwendar** [comment]
`cho 1 | sudo tee /sys/module/kvm/parameters/ignore_msrs` solved my problem, archlinux
